### PR TITLE
azure - autotag created date action & Tags tests refactor

### PR DESCRIPTION
**Pull request by logachev**
- Created 'auto-tag-date' action, similar to 'auto-tag-user'. Minor refactoring to share as much code between them as possible

 - Huge refactoring in Tags tests:
    - Split giant 'TestTags.py' file to a number of files: 1 per action or filter for unit-tests
    - Single file with all 'Functional' (tests that require real resource\cassette) for 'Tags' action and filter
    - Functional tests configure resources in class setup\teardown methods. This methods aren't recorded in cassettes.

All tests from old testtags.py have equivalent test in new unittests (with some bugs fix).

'Functional' tests check regular scenario to catch any regressions because of SDK\API changes.

'Unittests' don't leverage cassettes\real scenarios and it allows us to test as many edge-case scenarios, syntax etc as we want to

closes https://github.com/cloud-custodian/cloud-custodian/issues/4239

### CODE REVIEW

File: tools/c7n_azure/c7n_azure/actions/tagging.py [hunk 1]
```diff
@@ -1,18 +1,21 @@
 import datetime
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/actions/tagging.py]: Please add the Copyright headers

File: tools/c7n_azure/c7n_azure/actions/tagging.py [hunk 2]
```diff
@@ -179,16 +157,101 @@ def _process_resource(self, resource, event):
...
-        # issue tag action to label user
-        TagHelper.add_tags(self, resource, {self.tag_key: user})
+        # fetch activity logs
+        logs = self.client.activity_logs.list(
+            filter=query_filter,
+            select=self.query_select
+        )
+
+        # get the user who issued the first operation
+        operation_name = "%s/write" % resource_type
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/actions/tagging.py]: I discovered we could potentially be even smarter here. There is a substatus in the activity logs... the problem is I don't think we can guarantee that all RPs use the same text. However, something we can consider investigating.

```
    "subStatus": {
        "value": "Created",
        "localizedValue": "Created (HTTP Status Code: 201)"
    },
```

**Comment by thisisshi** [comment]
33k lines removed :0

**Comment by stefangordon** [comment]
Looking good.
